pkgs/tools/text/mml/default.nix

```
@@ -0,0 +1,59 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, installShellFiles
+, installShellCompletions ? stdenv.hostPlatform == stdenv.buildPlatform
+, installManPages ? stdenv.hostPlatform == stdenv.buildPlatform
+, gnupg
+, withCompiler ? true
+, withInterpreter ? true
+, withPgpCommands ? false
+, withPgpGpg ? false
+, withPgpNative ? false
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "mml";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "soywod";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-8CFLtmN7CfcXcImRcqIaohT3vWWVfWzEwXIXvmL8MsQ=";
+  };
+
+  cargoSha256 = "35VEYViJJjD5rvaCPRJJSm/8bOO4BZKLOnynH6AQf+o=";
+
+  nativeBuildInputs = lib.optional (installManPages || installShellCompletions) installShellFiles;
+
+  buildInputs = lib.optional withPgpGpg gnupg;
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [ ]
+    ++ lib.optional withCompiler "compiler"
+    ++ lib.optional withInterpreter "interpreter"
+    ++ lib.optional withPgpCommands "pgp-commands"
+    ++ lib.optional withPgpGpg "pgp-gpg"
+    ++ lib.optional withPgpNative "pgp-native";
+
+  postInstall = lib.optionalString installManPages ''
+    mkdir -p $out/man
+    $out/bin/mml man $out/man
+    installManPage $out/man/*
+  '' + lib.optionalString installShellCompletions ''
+    installShellCompletion --cmd mml \
+      --bash <($out/bin/mml completion bash) \
+      --fish <($out/bin/mml completion fish) \
+      --zsh <($out/bin/mml completion zsh)
+  '';
+
+  meta = with lib; {
+    description = "CLI to compile MML messages to MIME messages and interpret MIME messages as MML messages.";
```

Should not end in a period, https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes
```suggestion
    description = "CLI to compile MML messages to MIME messages and interpret MIME messages as MML messages";
```